examples/ignite_simple.py

```diff
@@ -0,0 +1,145 @@
+"""
+Optuna example that optimizes convolutional neural networks using PyTorch Ignite.
+
+In this example, we optimize the validation accuracy of hand-written digit recognition using
+PyTorch Ignite and MNIST. We optimize the neural network architecture as well as the optimizer
+configuration. As it is too time consuming to use the whole MNIST dataset, we here use a small
+subset of it.
+
+We have the following two ways to execute this example:
+
+(1) Execute this code directly.
+    $ python ignite_simple.py
+
+
+(2) Execute through CLI.
+    $ STUDY_NAME=`optuna create-study --direction maximize --storage sqlite:///example.db`
+    $ optuna study optimize ignite_simple.py objective --n-trials=100 --study $STUDY_NAME \
+      --storage sqlite:///example.db
+"""
+
+from __future__ import division
+from __future__ import print_function
+
+from ignite.engine import create_supervised_evaluator
+from ignite.engine import create_supervised_trainer
+from ignite.engine import Events
+from ignite.metrics import Accuracy
+import torch
+from torch import nn
+import torch.nn.functional as F
+from torch.optim import Adam
+from torch.optim import SGD
+from torch.utils.data import DataLoader
+from torch.utils.data import Subset
+from torchvision.datasets.mnist import MNIST
+from torchvision.transforms import Compose
+from torchvision.transforms import Normalize
+from torchvision.transforms import ToTensor
+
+import optuna
+
+
+EPOCHS = 10
+TRAIN_BATCH_SIZE = 64
+VAL_BATCH_SIZE = 1000
+N_TRAIN_EXAMPLES = 3000
+N_VALID_EXAMPLES = 1000
+
+
+class Net(nn.Module):
+    def __init__(self, trial):
+        # We optimize dropout rate in a convolutional neural network.
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
+        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
+        dropout_rate = trial.suggest_uniform('dropout_rate', 0, 1)
+        self.conv2_drop = nn.Dropout2d(p=dropout_rate)
+        self.fc1 = nn.Linear(320, 50)
+        self.fc2 = nn.Linear(50, 10)
+
+    def forward(self, x):
+        x = F.relu(F.max_pool2d(self.conv1(x), 2))
+        x = F.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))
+        x = x.view(-1, 320)
+        x = F.relu(self.fc1(x))
+        x = F.dropout(x, training=self.training)
+        x = self.fc2(x)
+        return F.log_softmax(x, dim=-1)
+
+
+def create_optimizer(trial, model):
```

Comment:
(Optional) In my experience, tuning the learning rate makes little sense in a simple binary classification task with a single machine. `optimizer = Adam(model.parameters(), lr=1e-3)` is good enough.

Reply:
That makes sense.
According to [the reference](https://pytorch.org/docs/stable/_modules/torch/optim/adam.html#Adam), `lr=1e-3` seems to be the default value.
I'll change it to `optimizer = Adam(model.parameters())`.

On the other hand, I'll keep the learning rate tuning for [`SGD`](https://pytorch.org/docs/stable/optim.html#torch.optim.SGD) because the learning rate is a required parameter.

Reply:
I'll remove `SGD` from the choices because it rarely outperformed the default setting of `Adam`.